src/plugins/intel_gpu/tests/unit/passes/reorder_reshape_permute.cpp

```diff
@@ -0,0 +1,372 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convolution_inst.h"
+#include "intel_gpu/graph/program.hpp"
+#include "permute_inst.h"
+#include "program_wrapper.h"
+#include "reshape_inst.h"
+#include "test_utils.h"
+
+using namespace cldnn;
+using namespace ::tests;
+
+TEST(opt_reorder_reshape_permute, no_reshape) {
+    auto& engine = get_test_engine();
+    auto in_layout = layout{ov::PartialShape({1, 2, 4, 6}), data_types::f16, format::bfyx};
+    auto input = engine.allocate_memory(layout{ov::PartialShape({1, 2, 4, 6}), data_types::f16, format::bfyx});
+    auto weight = engine.allocate_memory(layout{ov::PartialShape({3, 2, 1, 1}), data_types::f16, format::bfyx});
+    tests::set_random_values<ov::float16>(input);
+    tests::set_random_values<ov::float16>(weight);
+
+    topology topology;
+    topology.add(input_layout("input", in_layout));
+    topology.add(data("weight", weight));
+    topology.add(
+        convolution("convolution", input_info("input"), "weight", "", 1, {1, 1}, {1, 1}, {0, 0}, {0, 0}, false));
+    topology.add(reorder("reorder_inter", input_info("convolution"), format::bfyx, data_types::f16));
+    topology.add(permute("permute_inter", input_info("reorder_inter"), {0, 2, 3, 1}));
+    topology.add(softmax("softmax", input_info("permute_inter"), 1));
+    ExecutionConfig config = get_test_default_config(engine);
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(true));
+    config.set_property(ov::intel_gpu::optimize_data(true));
+    auto prog = program::build_program(engine, topology, config);
+
+    network net(prog);
+
+    net.set_input_data("input", input);
+    auto output = net.execute();
+
+    ExecutionConfig ref_config = get_test_default_config(engine);
+    ref_config.set_property(ov::intel_gpu::optimize_data(false));
+    cldnn::network ref_network(engine, topology, ref_config);
+    // reorder node is removed in primitive fusing
+    // later permute is optimized after convolution in selected preferred formats, e.g conv + permute
+    auto optimzed_nodes = net.get_program()->get_optimized();
+    auto it =
+        std::find_if(std::begin(optimzed_nodes), std::end(optimzed_nodes), [&](cldnn::program::optimized_info& oi) {
+            return oi.first == "reorder_inter";
+        });
+    ASSERT_NE(it, optimzed_nodes.end());
+    auto permute_inst = net.get_primitive("permute_inter");
+    if (net.get_primitive("convolution")->get_impl()->is_onednn()) {
+        ASSERT_TRUE(permute_inst->can_be_optimized());
+    }
+    auto out_mem = output.at("softmax").get_memory();
+    mem_lock<ov::float16> lock(out_mem, get_test_stream());
+
+    ref_network.set_input_data("input", input);
+    auto ref_output = ref_network.execute();
+    auto ref_out_mem = ref_output.at("softmax").get_memory();
+    mem_lock<ov::float16> lock_ref(ref_out_mem, get_test_stream());
+    auto tolerance = default_tolerance(ref_out_mem->get_layout().data_type);
+    for (size_t i = 0; i < out_mem->count(); i++) {
```

Comment:
Do we need check the expected exec_graph?

Reply:
yes, it is checking part of the exec graph, for example, the optimized_nodes and permute_inst above